plugins/woo-ai/src/components/tour-spotlight/tour-spotlight.tsx

```diff
@@ -1,71 +1,96 @@
 /**
  * External dependencies
  */
-import Tour, { TourStepRendererProps } from '@automattic/tour-kit';
-import { Button } from '@wordpress/components';
-import { useState } from '@wordpress/element';
-import { __ } from '@wordpress/i18n';
-import React from 'react';
+import { TourKit } from '@woocommerce/components';
+import { store as preferencesStore } from '@wordpress/preferences';
+import { useDispatch, select } from '@wordpress/data';
+import { useState, useEffect } from '@wordpress/element';
+import { Config } from '@automattic/tour-kit';
 
 type TourSpotlightProps = {
-	onDismiss: () => void;
+	id: string;
 	title: string;
 	description: string;
 	reference: string;
-	className?: string;
+	placement?: Config[ 'placement' ];
+	spotlightParent?: HTMLElement;
+	onDismissal?: () => void;
+	onStepViewOnce?: () => void;
 };
 
-export default function TourSpotlight( {
-	onDismiss,
+export const TourSpotlight: React.FC< TourSpotlightProps > = ( {
+	id,
 	title,
 	description,
 	reference,
-	className,
-}: TourSpotlightProps ) {
-	const [ showTour, setShowTour ] = useState( true );
+	placement = 'auto',
+	spotlightParent = document.body,
+	onDismissal = () => {},
+	onStepViewOnce = () => {},
+} ) => {
+	const preferenceId = `spotlightDismissed-${ id }`;
 
-	const handleDismiss = () => {
-		setShowTour( false );
-		onDismiss();
-	};
-	// Define a configuration for the tour, passing along a handler for closing.
-	const config = {
-		steps: [
-			{
-				referenceElements: {
-					desktop: reference,
-				},
-				meta: {
-					description,
-				},
-			},
-		],
-		renderers: {
-			tourStep: ( { currentStepIndex }: TourStepRendererProps ) => {
-				return (
-					<div className={ className }>
-						<h3>{ title }</h3>
-						<p>
-							{
-								config.steps[ currentStepIndex ].meta
-									.description
-							}
-						</p>
-						<Button onClick={ handleDismiss }>
-							{ __( 'Got it', 'woocommerce' ) }
-						</Button>
-					</div>
-				);
-			},
-			tourMinimized: () => <div />,
-		},
-		closeHandler: () => handleDismiss(),
-		options: {},
-	};
+	const anchorElement = document.querySelector( reference );
+	const hasBeenDismissedBefore = select( preferencesStore ).get(
+		'woo-ai-plugin',
+		preferenceId
+	);
+	const { set } = useDispatch( preferencesStore );
+	const [ isSpotlightVisible, setIsSpotlightVisible ] =
+		useState< boolean >( false );
+
+	// Avoids showing the spotlight before the layout is ready.
+	useEffect( () => {
```

Comment:
This is a bit weird, but it does eliminate some weirdness around spotlight placement being incorrect as images load.

Reply:
> This is a bit weird, but it does eliminate some weirdness around spotlight placement being incorrect as images load.

You're referring to the 250ms manual timeout here?
As things load on the page, yeah, the layout shift can cause problems.
Does it not work to listen to the `load` JS event rather than wait an arbitrary amount of time?

Reply:
Yeah, I'd thought about that, although that would need to be done in the parent component, and it adds a fair bit of complexity. And either way it might be a good idea to retain this delay for other use cases. Not sure whether it's worth the extra code, with that in mind. What do you think?

Reply:
If you've already thought about it and decided this approach was better, I trust your judgement. 👍